packages/react-select/src/Select.js

```diff
@@ -138,6 +138,8 @@ export type Props = {
   formatGroupLabel: typeof formatGroupLabel,
   /* Formats option labels in the menu and control as React components */
   formatOptionLabel?: (OptionType, FormatOptionLabelMeta) => Node,
+  /* Applies focus to the first option in the menu when opened */
+  focusDefaultOption: boolean,
```

Comment:
nit and bike shedding: I prefer `autoFocusOption` or `autoFocusFirstOption` as the prop name, but that's just a personal preference.

1. `autoFocus` vs `focus`: in my mind reinforces that this is something that happens once when the menu opens, but that the focus can change with user input.
2. `option` or `firstOption` vs `defaultOption`: `defaultOption` makes it sound like to me that the the default option is something that is customizable when it's not. Using `option` still leaves room for future props that could change which `option` is focused by default, but that's not the case at the moment.

But I'm fine with `focusDefaultOption` if that's easier.

Reply:
Hmmm I thought `autoFocus` would be like [input#autofocus](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefautofocus) where the field snatches a user's focus.

Reply:
Yeah, there's a little bit of overloading with the word "focus" which makes it hard to differentiate, since we have to talk about both focus of the Select component and the focus of individual options when the menu is open.

I think maybe the most descriptive (but probably too long) prop name would be `autoFocusFirstOptionOnMenuOpen`  where the `onMenuOpen` disambiguates when the auto-focusing would happen.

Reply:
Knowing the full context, I prefer your `autoFocusOption`; however, could we not change the type signature to allow people to choose a different item to receive first focus? Who's to say maybe somebody would not want to focus the last option, or some middle one 🤷 . Regardless, I like that naming best. If it will be constrained to just boolean, I like your longer variable name: `autoFocusFirstOptionOnMenuOpen` - very crystal clear.

Reply:
Yeah, it's hard to know where to stop as far as customization. The options ordered from most customizable to least customizable are:

1. Provide a `getDefaultFocusedOption` prop which is a function where the user is passed information and they can return the option they want to be focused by default.
2. Make `autoFocusOption` a prop that can be `first`, `last`, or false (can be expanded later, but I'm not sure what else you would need.
3. Make `autoFocusOption` a prop that can be `true` or `false`.

Since `react-select` is already very complex and hard to keep straight with all its edge cases, I think we want to be conservative and avoid complexities.

That's why I suggested `autoFocusOption` instead of `autoFocusFirstOption` as my top choice (if we go with `true | false` because it leaves open the possibility of adding `'first' | 'last'` options later on.

But you make a good point. Maybe `autoFocusOptionOnMenuOpen` is the best of both worlds.